Allowing multiple accounts to be set in the SQLSYSADMINACCOUNTS parameter when installing sql server

jerrelblankenship opened the pull request:
Fixed issue where you could not submit multiple accounts to the SQLSYSADMINACCOUNTS parameter when installing sql server

chef-supermarket commented:
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

jerrelblankenship commented:
@pburkholder You and I just talked about this pull request. Thank you for taking a look at this.

pburkholder commented:
@jerrelblankenship ACK your PR. I'm still digging out from ChefConf, and will return to this this week.

jerrelblankenship commented:
Hi @pburkholder , I was wondering if there was any update on this pull request. I have signed the CLA, but it still states that I need to sign one.

irvingpop commented:
hey @jerrelblankenship the problem is that there's a mismatch between the email address in the commit message  ( see:  https://help.github.com/articles/setting-your-email-in-git/ ) and the one associated to your github account.   

So you'll either need to associate your work email address with your github account, or use your personal email address in your git commit messages.

jerrelblankenship commented:
Hi @irvingpop  I associated my work email address with my github account. The work email is not the primary email account if that makes a difference.

irvingpop commented:
hi @btm could you please poke Currybot when you have a chance?

chef-supermarket commented:
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

btm commented:
Done! :D

irvingpop commented:
thank you!!

irvingpop commented:
Hi @jerrelblankenship - while we are here, some code review :)

`default['sql_server']['sysadmins']` is currently a `String` and in this PR you're redefining it to an `Array`, or at least that's implied because you're using `join`.    I suggest two changes to make this help future users and not break existing users:

1.  Change the attribute default value to be an array like `[ 'Administrator' ]`
2.  make `sql_sys_admin_list` smart enough to know if it's taking an array (new behavior) or a string (old behavior)

jerrelblankenship commented:
Hi @irvingpop, I have made the changes you suggested. Thank you for the feedback.

jerrelblankenship commented:
Hi @irvingpop sorry for all the back and forth. I have made the changes you suggested.

irvingpop commented:
Hey @jerrelblankenship the logic is still backward on this.   `join` is only a valid method on an Array, but the way you have it now it would do a it would do a join if the value is *not* an array.

```ruby
if node['sql_server']['sysadmins'].is_a? Array
  sql_sys_admin_list = node['sql_server']['sysadmins']
else  
  sql_sys_admin_list = node['sql_server']['sysadmins'].join(' ')
end
```

jerrelblankenship commented:
I fixed the bug and added ChefSpec tests to the cookbook with help from @irvingpop. Here is the output from the ChefSpec run
```
$ rspec
......

Finished in 1.54 seconds (files took 3.69 seconds to load)
6 examples, 0 failures

ChefSpec Coverage report generated...

  Total Resources:   7
  Touched Resources: 1
  Touch Coverage:    14.29%

Untouched Resources:

  template[\var\folders\sq\clk76llj1pb5t2hkrs7mxcjc0000gn\T\d20150514-7576-v6jw6k\ConfigurationFile.ini]   sql_server/recipes/server.rb:44
  windows_package[Microsoft SQL Server 2008 R2 (64-bit)]   sql_server/recipes/server.rb:51
  windows_service[MSSQL$SQLEXPRESS]   sql_server/recipes/server.rb:60
  registry_key[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SQL Server\MSSQL10_50.SQLEXPRESS\MSSQLServer\SuperSocketNetLib\Tcp\IPAll]   sql_server/recipes/server.rb:65
  gem_package[tiny_tds]              sql_server/recipes/client.rb:45
  template[\var\folders\sq\clk76llj1pb5t2hkrs7mxcjc0000gn\T\d20150514-7576-oy29w\ConfigurationFile.ini]   sql_server/recipes/server.rb:44
```

irvingpop commented:
great work on this!  :+1:
